### ISSUE TITLE
SC2241: allow double dash with exit; `exit -- 1` works just fine

### ISSUE THREAD

**Issue by SebastianSimon** [koalaman/shellcheck#2044]
## Bug

- Rule Id: SC2241
- My shellcheck version: online
- [x] The rule's wiki page does not already cover this
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

### Here’s a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
exit -- 1
```

### Here’s what shellcheck currently says:

```none
Line 2:
exit -- 1
     ^-- SC2241: The exit status can only be one integer 0-255. Use stdout for other data.
```

### Here’s what I wanted or expected to see:

```
No issues detected!
```

Either remove that error, or, if this is actually hazardous in some way (e.g. not portable), replace it with a more specific warning.

### Rationale

I’m trying to use `--` whenever possible and with `exit` it seems to work just fine in Bash:

```sh
$ bash
$ exit 42
exit
$ echo $?
42
$ bash
$ exit -- 66
exit
$ echo $?
66
```

**Comment by kvantour** [comment]
The Posix Standard makes the following comment on shell-build-in functions:

> Some of the special built-ins are described as conforming to XBD Utility Syntax Guidelines. For those that are not, the requirement in Utility Description Defaults that "--" be recognized as a first argument to be discarded does not apply and a conforming application shall not use that argument.
>
> <sub>**source:** https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_14</sub>

The `exit` built-in does not support  the XBD Utility Syntax Guidelines, and therefore should not use this argument.

The Bash Manual makes the following statement:

> Many of the builtins have been extended by POSIX or Bash.
>
> Unless otherwise noted, each builtin command documented as accepting options preceded by ‘-’ accepts ‘--’ to signify the end of the options. The `:`, `true`, `false`, and `test/[` builtins do not accept options and do not treat ‘--’ specially. The `exit`, `logout`, `return`, `break`, `continue`, `let`, and `shift` builtins accept and process arguments beginning with ‘-’ without requiring ‘--’. Other builtins that accept arguments but are not specified as accepting options interpret arguments beginning with ‘-’ as invalid options and require ‘--’ to prevent this interpretation.
>
> <sub>**source:** https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Shell-Builtin-Commands</sub>

**In summary:**
The command `exit -- [n]` is invalid syntax in Posix SH. In bash, `exit` does not require it, so I believe it should not be accepted. 

**Note:** I could misinterpret what is written in those standards.
